sql: fix minor typos/bug in updateZoneConfigsForAllTables

otan opened the pull request:
Noticed this implementing something else for REGIONAL BY ROW.

* Fixed a bug where LookupFlags AvoidCache field was reset between
  schema jumps.
* Fixed a typo where the comment for updateZoneConfigsForAllTables was
  not updated.
* Refactored logic for iterating over each table to be a separate
  function for re-usability.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59124)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
Nice change!  Only nits below.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59124#-:-MRRN0Fd8dGY2YN1KULZ:b4z593m)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/region_util.go, line 344 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59124#-MRRLDzd-Wme6nC1aCEB:-MRRLDzd-Wme6nC1aCEC:bvkc8fh) ([raw file](https://github.com/cockroachdb/cockroach/blob/45ac72050c6dfea583d3974bf8baf414b498700a/pkg/sql/region_util.go#L344)):*
> ```Go
> }
> 
> // forEachLocalityConfiguredTableIndatabase loops through each schema and table
> ```

Nit: typos

-  uppercase D in database
- name in comment doesn't match function name (didn't before either, which is my bad)
___
*[pkg/sql/region_util.go, line 376 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59124#-MRRMB2s66RQQmCShrW2:-MRRMB2s66RQQmCShrW3:b6zrrk4) ([raw file](https://github.com/cockroachdb/cockroach/blob/19addd018c462875515f94c5d77a3848f2995443/pkg/sql/region_util.go#L376)):*
> ```Go
> 			return err
> 		}
> 		lookupFlags.Required = false
> ```

Good catch!


<!-- Sent from Reviewable.io -->

otan committed:
sql: fix minor typos/bug in updateZoneConfigsForAllTables

Noticed this implementing something else for REGIONAL BY ROW.

* Fixed a bug where LookupFlags AvoidCache field was reset between
  schema jumps.
* Fixed a typo where the comment for updateZoneConfigsForAllTables was
  not updated.
* Refactored logic for iterating over each table to be a separate
  function for re-usability.

Release note: None

otan commented:
thanks!

changes made

bors r=ajstorm
